wordpress-mobile/WordPress-iOS
# Fixes the Media `featuredOnPosts` relationship

Fixes #9295.

A crash that was previously reported in https://github.com/wordpress-mobile/WordPress-iOS/issues/9182 and fixed in https://github.com/wordpress-mobile/WordPress-iOS/pull/9209 [resurfaced](https://github.com/wordpress-mobile/WordPress-iOS/issues/9295) due to (seemingly) a bad merge between two data model updates, which knocked out a fix to the Media entity.

We recently added a relationship to Media for featured images: `featuredOnPost`. The relationship originally referenced `Post` instead of `AbstractPost`, which meant that it couldn't handle Pages, and so caused a crash. 

As a result of the data model merge, we've seen a few instances in local testing where the app has been unable to migrate to the latest model. This PR fixes the relationship, and also adds a test to ensure that a Media object can be migrated successfully.

**To test:**

* Build the app, check the tests run!
* If you like, try installing over `release/9.9` and check the app is able to migrate.

cc @elibud @diegoreymendez
